### ISSUE TITLE
GM can't see what players do in instances (BB #740)

### ISSUE THREAD

**Issue by OregonAsari** [OregonCore/OregonCore#740]
This issue was migrated from bitbucket.
**Original Reporter:** PadreWoW
**Original Date:** 12.11.2014 17:37:45 GMT+0000
**Original Priority:** major
**Original Type:** bug
**Original State:** resolved
**Direct Link:** https://bitbucket.org/oregon/oregoncore/issues/740

<hr>

When GM use .goname xxx when xxx is in instance, GM teleport to instance, but can't see anyone

**Comment by OregonAsari** [comment]
This comment was migrated from bitbucket.
**Original User:** PadreWoW
**Original Date:** 12.11.2014 17:40:45 GMT+0000
**Direct Link:** https://bitbucket.org/oregon/oregoncore/issues/740/#comment-13601762

<hr>

the problem is - GM teleports to another copy of instance

**Comment by OregonAsari** [comment]
This comment was migrated from bitbucket.
**Original User:** PadreWoW
**Original Date:** 12.11.2014 17:53:20 GMT+0000
**Direct Link:** https://bitbucket.org/oregon/oregoncore/issues/740/#comment-13602008

<hr>

I think the problem is here:

https://bitbucket.org/oregon/oregoncore/commits/3cdbafb1e55100c00440a8d0ec77c237a35064f2

**Comment by OregonAsari** [comment]
This comment was migrated from bitbucket.
**Original User:** PadreWoW
**Original Date:** 12.11.2014 18:20:31 GMT+0000
**Direct Link:** https://bitbucket.org/oregon/oregoncore/issues/740/#comment-13602388

<hr>

Yep. The problem is here.

There should be this:

```
#!c++

// if the player or the player's group is bound to another instance
        // the player will not be bound to another one
        InstancePlayerBind *pBind = _player->GetBoundInstance(target->GetMapId(), target->GetDifficulty());
        if (!pBind)
        {
        Group *group = _player->GetGroup();
        // if no bind exists, create a solo bind
        InstanceGroupBind *gBind = group ? group->GetBoundInstance(target) : NULL;                // if no bind exists, create a solo bind
        if (!gBind)
         if (InstanceSave *save = sInstanceSaveMgr.GetInstanceSave(target->GetInstanceId()))
         _player->BindToInstance(save, !save->CanReset());
        }

            _player->SetDifficulty(target->GetDifficulty());
        }
```

**Comment by OregonAsari** [comment]
This comment was migrated from bitbucket.
**Original User:** LordUsagi
**Original Date:** 12.11.2014 22:11:57 GMT+0000
**Direct Link:** https://bitbucket.org/oregon/oregoncore/issues/740/#comment-13606783

<hr>

[Core/Commands] Prevent entering on a clean bind.
                Fixes #740.

→ <<cset d50a0883f2b8>>
